### PR TITLE
ls: ignore leading period when sorting by name

### PR DESCRIPTION
**Pull request by ricardoaiglesias**
ls now behaves like GNU ls with respect to sorting files by ignoring
leading periods when sorting by main.

Added tests to ensure "touch a .a b .b ; ls" returns ".a  a  .b  b"

Note: The non `src/ls/` files that were modified were modified because of clippy warnings that wouldn't let me commit otherwise.

### CODE REVIEW

**Comment by ricardoaiglesias** [comment]
Every check except  `stable x86_64-unknown-freebsd-12` passes. The failing check fails on `cp`, which was not modified during this PR. I think it's ready to merge. 
Thoughts?

File: src/uu/ls/src/ls.rs [hunk 1]
```diff
@@ -1106,7 +1106,17 @@ fn sort_entries(entries: &mut Vec<PathData>, config: &Config) {
             entries.sort_by_key(|k| Reverse(k.md.as_ref().map(|md| md.len()).unwrap_or(0)))
         }
         // The default sort in GNU ls is case insensitive
-        Sort::Name => entries.sort_by_key(|k| k.file_name.to_lowercase()),
+        Sort::Name => entries.sort_by_key(|k| {
+            let has_dot: bool = k.file_name.starts_with('.');
+            let filename_nodot: String = if has_dot {
```
Comment (siebenHeaven) [inline src/uu/ls/src/ls.rs:1111]: Names are usually short so this maybe a nitpick, but we could work with something like this that does not involve clone / collect:
```
let filename_nodot = &k.file_name[if has_dot { 1 } else { 0 }..];
```

File: src/uu/printf/src/tokenize/num_format/formatters/base_conv/mod.rs [hunk 2]
```diff
@@ -158,7 +158,7 @@ pub fn base_conv_float(src: &[u8], radix_src: u8, _radix_dest: u8) -> f64 {
     // to implement this for arbitrary string input.
     // until then, the below operates as an outline
     // of how it would work.
-    let result: Vec<u8> = vec![0];
+    let _result: Vec<u8> = vec![0];
```
Comment (sylvestre) [inline src/uu/printf/src/tokenize/num_format/formatters/base_conv/mod.rs:161]: please remove this unrelated change :)

File: src/uu/ls/src/ls.rs (hunk 1 above)
Comment (sylvestre) [reply src/uu/ls/src/ls.rs:1111]: And please add a test to cover this case :)

**Review by sylvestre — CHANGES_REQUESTED** [review]
Looks great :)

File: src/uu/ls/src/ls.rs (hunk 1 above)
Comment (ricardoaiglesias) [reply src/uu/ls/src/ls.rs:1111]: I think I did the clone/collect thing because I was thinking about the general case of "characters in Rust can be up to four bytes long", but I guess if we know the first character is an ASCII `.`, then we're free to slice it.
